pkg/api/validation/validation.go

```diff
@@ -1309,6 +1309,19 @@ func validateStorageOSPersistentVolumeSource(storageos *api.StorageOSPersistentV
 	return allErrs
 }
 
+func validateCSIPersistentVolumeSource(csi *api.CSIPersistentVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(csi.Driver) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("driver"), ""))
+	}
+
+	if len(csi.VolumeHandle) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("volumeHandle"), ""))
+	}
+
```

Comment:
Also need to validate `MountSecretRef` and `AttachSecretRef` if they are not nil.

Like, 
```
if csi.MountSecretRef != nil {
		if len(csi.MountSecretRef.Name) == 0 {
			allErrs = append(allErrs, field.Required(fldPath.Child("MountSecretRef", "name"), ""))
		}
		...
}
```

Reply:
@dixudx ok good catch. Will look into this as well.  Thanks.

Reply:
@dixudx I dbl checked, `MountSecretRef` is no longer part of the API type.